Formula/binutils.rb

```diff
@@ -20,30 +20,46 @@ class Binutils < Formula
 
   uses_from_macos "zlib"
 
+  on_linux do
+    depends_on "glibc@2.13" => :build
+  end
+
   def install
-    system "./configure", "--disable-debug",
-                          "--disable-dependency-tracking",
-                          "--enable-deterministic-archives",
-                          "--prefix=#{prefix}",
-                          "--infodir=#{info}",
-                          "--mandir=#{man}",
-                          "--disable-werror",
-                          "--enable-interwork",
-                          "--enable-multilib",
-                          "--enable-64-bit-bfd",
-                          "--enable-gold",
-                          "--enable-plugins",
-                          "--enable-targets=all",
-                          "--with-system-zlib",
-                          "--disable-nls"
+    # Fix: error: 'LONG_MIN' undeclared
+    ENV["HOMEBREW_INCLUDE_PATHS"] = nil if OS.linux?
```

Comment:
Is this something we can report upstream?

Reply:
I think this is related to the behaviour of our compiler shims, and not anything that can/should be fixed upstream.

Reply:
It'd be good to detail a bit more in the comment here and potentially move this logic into Homebrew/brew.

Reply:
Maybe it has something to do with [this article](https://github.com/cirosantilli/notes/blob/master/stack-overflow/binutils-long-min.md)?

Reply:
Is this because `HOMEBREW_INCLUDE_PATHS` is adding back host directories like `/usr/include`?  I'll try to look at this when I get a chance.  Also since this is Linux only, I think the `if OS.linux?` is unneeded.

Reply:
Here is the full error without the workaround:
```
if [ x"" != x ]; then \
  gcc-5 -c -DHAVE_CONFIG_H -g -O2      -I. -I./../include  -W -Wall -Wwrite-strings -Wc++-compat -Wstrict-prototypes -pedantic  -D_GNU_SOURCE    ./floatformat.c -o noasan/floatformat.o; \
else true; fi
./fibheap.c: In function ‘fibheap_replace_key_data’:
./fibheap.c:38:24: error: ‘LONG_MIN’ undeclared (first use in this function)
 #define FIBHEAPKEY_MIN LONG_MIN
                        ^
./fibheap.c:220:30: note: in expansion of macro ‘FIBHEAPKEY_MIN’
   if (okey == key && okey != FIBHEAPKEY_MIN)
                              ^
./fibheap.c:38:24: note: each undeclared identifier is reported only once for each function it appears in
 #define FIBHEAPKEY_MIN LONG_MIN
                        ^
./fibheap.c:220:30: note: in expansion of macro ‘FIBHEAPKEY_MIN’
   if (okey == key && okey != FIBHEAPKEY_MIN)
                              ^
./fibheap.c: In function ‘fibheap_delete_node’:
./fibheap.c:38:24: error: ‘LONG_MIN’ undeclared (first use in this function)
 #define FIBHEAPKEY_MIN LONG_MIN
                        ^
./fibheap.c:261:36: note: in expansion of macro ‘FIBHEAPKEY_MIN’
   fibheap_replace_key (heap, node, FIBHEAPKEY_MIN);
                                    ^
if [ x"" != x ]; then \
  gcc-5 -c -DHAVE_CONFIG_H -g -O2      -I. -I./../include  -W -Wall -Wwrite-strings -Wc++-compat -Wstrict-prototypes -pedantic  -D_GNU_SOURCE    ./fnmatch.c -o noasan/fnmatch.o; \
else true; fi
gcc-5 -c -DHAVE_CONFIG_H -g -O2      -I. -I./../include  -W -Wall -Wwrite-strings -Wc++-compat -Wstrict-prototypes -pedantic  -D_GNU_SOURCE  ./floatformat.c -o floatformat.o
gcc-5 -c -DHAVE_CONFIG_H -g -O2      -I. -I./../include  -W -Wall -Wwrite-strings -Wc++-compat -Wstrict-prototypes -pedantic  -D_GNU_SOURCE  ./fnmatch.c -o fnmatch.o
Makefile:773: recipe for target 'fibheap.o' failed
```

Reply:
```
gcc-5 called with: -c -DHAVE_CONFIG_H -g -O2 -I. -I./../include -W -Wall -Wwrite-strings -Wc++-compat -Wstrict-prototypes -pedantic -D_GNU_SOURCE ./fibheap.c -o fibheap.o
superenv removed:  -g -W -Wall -Wwrite-strings -Wc++-compat -Wstrict-prototypes -pedantic
superenv added:    -pipe -w -march=native -isystem/home/linuxbrew/.linuxbrew/include -I/home/linuxbrew/.linuxbrew/opt/glibc@2.13/include
superenv executed: gcc-5 -pipe -w -O2 -march=native -c -DHAVE_CONFIG_H -I. -I./../include -D_GNU_SOURCE ./fibheap.c -o fibheap.o -isystem/home/linuxbrew/.linuxbrew/include -I/home/linuxbrew/.linuxbrew/opt/glibc@2.13/include
```

Reply:
And here is the supernev log with the workaround:
```
gcc-5 called with: -c -DHAVE_CONFIG_H -g -O2 -I. -I./../include -W -Wall -Wwrite-strings -Wc++-compat -Wstrict-prototypes -pedantic -D_GNU_SOURCE ./fibheap.c -o fibheap.o
superenv removed:  -g -W -Wall -Wwrite-strings -Wc++-compat -Wstrict-prototypes -pedantic
superenv added:    -pipe -w -march=native -isystem/home/linuxbrew/.linuxbrew/include
superenv executed: gcc-5 -pipe -w -O2 -march=native -c -DHAVE_CONFIG_H -I. -I./../include -D_GNU_SOURCE ./fibheap.c -o fibheap.o -isystem/home/linuxbrew/.linuxbrew/include
```